champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/brute/data/BlockTossObject.java

```diff
@@ -309,15 +320,12 @@ public void impact(Player caster) {
             final EntityProperty relation = nearbyEntry.getValue();
 
             if (relation != EntityProperty.FRIENDLY) {
-                if (!ent.hasLineOfSight(impactLocation)) continue;
                 // Damage anybody who is not friendly
-                damaged.add(ent);
-                Vector knockback = ent.getLocation().toVector().subtract(impactLocation.toVector());
-                final double strength = (radius * radius - ent.getLocation().distanceSquared(impactLocation)) / (radius * radius);
-                VelocityData velocityData = new VelocityData(knockback, strength, false, 0.0, 0.0, 3.0, true);
-                UtilVelocity.velocity(ent, caster, velocityData);
-                UtilDamage.doCustomDamage(new CustomDamageEvent(ent, caster, null, EntityDamageEvent.DamageCause.CUSTOM, damage, false, skill.getName()));
-                UtilMessage.simpleMessage(ent, skill.getName(), "<alt2>%s</alt2> hit you with <alt>%s</alt>.", caster.getName(), skill.getName());
+                if(!ent.hasLineOfSight(impactLocation)) continue;
+                if(!alreadyHit) {
```

Comment:
Just remove the directly impacted entity from `nearby` so you don't have to check if the entities are already hit.

Reply:
since I already use this boolean system in previous parts of the logic without access to the nearby map that contains direct hit player, id like to keep it here.